### DISCUSSION TITLE
Ability to add browser extensions

### DISCUSSION THREAD

**Discussion by 013nour** [unclecode/crawl4ai#547] [Feature requests]
Thanks for the great work and effort, 

one suggestion in mind to allow adding extensions to the browser on creating it, can be as easy as just passing the extension path. 

also, a session of crawling can start health and normal but when error encountered the returned object would not have any data? anything that can be done to avoid this? 

thanks!

**Comment by unclecode** [comment]
@013nour Thx for trying our library. I started from the second one. Would you please give me an example of the error that you've faced? The proposed behavior is to return the result with the flag of success set to false, then the details of the error message, and you can understand why that happened. If you give me an example, I will give you a better explanation about the first suggestion. very interesting idea, I'll definitely put that in our backlog. That gives our library a very unique reason why it should be used. Thank you for sharing that one.
